workers/unity/Packages/io.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs

```diff
@@ -8,50 +11,63 @@ namespace Improbable.Gdk.TransformSynchronization
     [UpdateInGroup(typeof(FixedUpdateSystemGroup))]
     public class DefaultApplyLatestTransformSystem : ComponentSystem
     {
-        private EntityQuery rigidbodyGroup;
-        private EntityQuery transformGroup;
+        private ComponentType[] baseComponentTypes;
+        private EntityQuery transformQuery;
+
+        private readonly Dictionary<Type, Action> applyLatestTransformActions = new Dictionary<Type, Action>();
 
         protected override void OnCreate()
         {
             base.OnCreate();
 
-            rigidbodyGroup = GetEntityQuery(
-                ComponentType.ReadOnly<Rigidbody>(),
+            baseComponentTypes = new[]
+            {
                 ComponentType.ReadOnly<TransformToSet>(),
                 ComponentType.ReadOnly<SetTransformToGameObjectTag>(),
                 ComponentType.ReadOnly<TransformInternal.ComponentAuthority>()
-            );
-            rigidbodyGroup.SetFilter(TransformInternal.ComponentAuthority.NotAuthoritative);
+            };
 
-            transformGroup = GetEntityQuery(
-                ComponentType.Exclude<Rigidbody>(),
-                ComponentType.ReadOnly<UnityEngine.Transform>(),
-                ComponentType.ReadOnly<TransformToSet>(),
-                ComponentType.ReadOnly<SetTransformToGameObjectTag>(),
-                ComponentType.ReadOnly<TransformInternal.ComponentAuthority>()
-            );
-            transformGroup.SetFilter(TransformInternal.ComponentAuthority.NotAuthoritative);
+            RegisterTransformSyncType(new RigidbodyTransformSync());
+        }
+
+        internal void RegisterTransformSyncType<T>(ITransformSync<T> impl)
+            where T : class
+        {
+            var entityQuery = GetEntityQuery(TransformUtils.ConstructEntityQueryDesc<T>(baseComponentTypes));
+            entityQuery.SetFilter(TransformInternal.ComponentAuthority.NotAuthoritative);
+
+            applyLatestTransformActions.Add(typeof(T), () => Entities.With(entityQuery).ForEach((EntityQueryBuilder.F_DC<TransformToSet, T>) impl.ApplyLatestTransform));
```

Comment:
can we make this not one line 😬

Reply:
I'm actually pretty sure I ran the linter over this 🤔 

Let me try again